### PR TITLE
Update dev instructions for using IntelliJ IDEA

### PR DESCRIPTION
**Pull request by romseygeek**
IntelliJ doesn't correctly pick up antlr generated sources just by examining 
poms, so needs a bit of manual intervention.  Adds some developer
documentation explaining how to do this.

### CODE REVIEW

File: devs/docs/basics.rst [hunk 1]
```diff
@@ -70,14 +70,28 @@ If you want to attach a debugger to the instance, you'd have to enable the debug
 Then create an "Attach" configuration in your IDE or editor. If using Visual
 Studio Code you can use the pre-defined "Attach to CrateDB" configuration.
 
+Using an IDE
+============
+
+`IntelliJ IDEA`_ will pick up project structure from the maven poms, but you will need to configure
```
Comment (matriv) [inline devs/docs/basics.rst:76]: To import the project into `IntelliJ IDEA`_, first, build the sources outside of your IDE with::

    $ ./mvnw clean install -DskipTests=true

and then import the project by selecting the root `pom.xml`.  `IntelliJ IDEA`_ will pick up project structure from the maven poms, but you might need to configure the 'crate-libs-sql-parser' generated sources manually, due to an `IDE bug`_.  Go to Project Settings / Modules / crate-libs-sql-parser and mark '>target/generated-sources/antlr' as a Source Folder.

Finally, run a `git clean` to remove pointless IntelliJ file changes to checked in config files.

**Review by matriv — COMMENTED** [review]
Thank you! I've left a suggestion with a slight rephrasing.

I've tried myself a fresh import from scratch and the generated sources are marked correctly without manual action, but there could be something cached in my intellij config.

File: devs/docs/basics.rst (hunk 1 above)
Comment (mfussenegger) [reply devs/docs/basics.rst:76]: There are btw. more generated sources. `pgwire` also uses antlr for the PostgreSQL text format array parser and `server` uses it for type signatures.

**Comment by romseygeek** [comment]
> I've tried myself a fresh import from scratch and the generated sources are marked correctly without manual action, but there could be something cached in my intellij config, from the time we used gradle.

That's interesting, which version of IntelliJ are you running?  I have 2023.2.5 CE

**Comment by romseygeek** [comment]
After some experimenting, it turns out that IntelliJ will pick up the generated sources correctly if we do a maven build first.  I'll update.

File: devs/docs/basics.rst [hunk 2]
```diff
@@ -70,14 +70,26 @@ If you want to attach a debugger to the instance, you'd have to enable the debug
 Then create an "Attach" configuration in your IDE or editor. If using Visual
 Studio Code you can use the pre-defined "Attach to CrateDB" configuration.
 
+Using an IDE
+============
+
+To import the project into `IntelliJ IDEA`_, first build the sources outside of your IDE with:
```
Comment (matriv) [inline devs/docs/basics.rst:76]: You can use the double colon, and check preview to see how it renders:
```suggestion
To import the project into `IntelliJ IDEA`_, first build the sources outside of your IDE with::
```

File: devs/docs/basics.rst [hunk 3]
```diff
@@ -70,14 +70,26 @@ If you want to attach a debugger to the instance, you'd have to enable the debug
...
+
+    $ ./mvnw clean install -DskipTests=true
+
+Then import the project into IntelliJ.  You may need to explicitly reload maven projects
+to ensure that it correctly picks up generated source files from antlr.
+
+Finally, run a `git clean` to remove pointless IntelliJ file changes to checked in config files
```
Comment (matriv) [inline devs/docs/basics.rst:83]:
```suggestion
Finally, run a `git clean` to remove pointless IntelliJ file changes to checked in config files.
```

**Review by matriv — APPROVED** [review]
LGTM, thx! left 2 small comments

**Comment by romseygeek** [comment]
@Mergifyio squash title+body

**Comment by romseygeek** [comment]
Closing in favour of #15143

**Pull request closed without merging**